gobblin-data-management/src/main/java/gobblin/data/management/trash/ProxiedTrash.java

```diff
@@ -0,0 +1,160 @@
+   * @return true if the move succeeded.
+   * @throws IOException
+   */
+  public boolean moveToTrashAsOwner(Path path) throws IOException {
+    String owner = this.fs.getFileStatus(path).getOwner();
+    return moveToTrashAsUser(path, owner);
+  }
+
+  /**
+   * Create a trash snapshot as the specified user.
+   * @param user user to proxy.
+   * @throws IOException
+   */
+  public void createTrashSnapshotAsUser(String user) throws IOException {
+    getUserTrash(user).createTrashSnapshot();
+  }
+
+  /**
+   * Purge trash snapshots as the specified user.
+   * @param user user to proxy.
+   * @throws IOException
+   */
+  public void purgeTrashSnapshotsAsUser(String user) throws IOException {
+    getUserTrash(user).purgeTrashSnapshots();
+  }
+
+  /**
+   * Create trash snapshots for all users with trash directories. These users are determined by listing all directories in
+   * the file system matching the trash pattern given by {@link #TRASH_LOCATION_KEY}.
+   * @throws IOException
+   */
+  public void createTrashSnapshotsForAllUsers() throws IOException {
+    for(String user : getAllUsersWithTrash()) {
+      createTrashSnapshotAsUser(user);
+    }
+  }
+
+  /**
+   * Purge trash snapshots for all users with trash directories. These users are determined by listing all directories in
+   * the file system matching the trash pattern given by {@link #TRASH_LOCATION_KEY}.
+   * @throws IOException
+   */
+  public void purgeTrashSnapshotsForAllUsers() throws IOException {
+    for(String user : getAllUsersWithTrash()) {
+      purgeTrashSnapshotsAsUser(user);
+    }
+  }
+
+  /**
+   * Find all users with trash directories by listing all directories in
+   * the file system matching the trash pattern given by {@link #TRASH_LOCATION_KEY}.
+   * @return List of users with trash directory.
+   * @throws IOException
+   */
+  protected List<String> getAllUsersWithTrash() throws IOException {
+    Path trashLocationGlob = new Path(this.properties.getProperty(TRASH_LOCATION_KEY).replaceAll("\\$USER", "*"));
+    Pattern userPattern = Pattern.compile(this.properties.getProperty(TRASH_LOCATION_KEY).replaceAll("\\$USER", "([^/])"));
+
+    List<String> users = Lists.newArrayList();
+    for(FileStatus fileStatus : this.fs.globStatus(trashLocationGlob)) {
+      Matcher matcher = userPattern.matcher(fileStatus.getPath().toString());
+      if(matcher.find()) {
+        users.add(matcher.group(1));
+      }
+    }
+
+    return users;
+  }
+
+  /**
+   * Get {@link gobblin.data.management.trash.Trash} instance for the specified user.
+   * @param user user for whom {@link gobblin.data.management.trash.Trash} should be generated.
+   * @return {@link gobblin.data.management.trash.Trash} as generated by proxied user.
+   * @throws IOException
+   */
+  protected Trash getUserTrash(final String user) throws IOException {
+    if(UserGroupInformation.getCurrentUser().getUserName().equals(user)) {
```

Comment:
Have you verified if `UserGroupInformation.getCurrentUser().getUserName()` simply returns the `[user-name]` or  `[user-name]/[cluster-name]@[realm]`?